api/orchestration_movers.go

```diff
@@ -115,6 +109,35 @@ func (o *datasyncOrchestrator) datamoverCreate(ctx context.Context, group string
 	return task, nil
 }
 
+// datamoverDelete deletes a data mover and all of its associated components
+func (o *datasyncOrchestrator) datamoverDelete(ctx context.Context, group, name string) error {
+	log.Infof("deleting data mover %s", name)
+
+	// get information about the datasync task
+	mover, err := o.datamoverDescribe(ctx, group, name)
+	if err != nil {
+		return err
+	}
+
+	// delete task
+	if _, err = o.datasyncClient.DeleteDatasyncTask(ctx, &datasync.DeleteTaskInput{
```

Comment:
Honestly I'm not sure about best practice but IMO you should initialize `err` in the scope of the conditional for these.  It's probably safe not to, but seems like it could lead to bugs in the future.